cmd/maddy-pam-helper/main.c

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <security/pam_appl.h>
+
+/*
+I really doublt it is a good idea to bring Go to the binary whose primary task
+is to call libpam using CGo anyway.
+*/
+
+struct pam_response *reply;
+
+int conv_func(int num_msg, const struct pam_message **msg, struct pam_response **resp, void *appdata_ptr) {
+    *resp = reply;
+    return PAM_SUCCESS;
+}
+
+int run() {
```

Comment:
In C functions without arguments are written `int run(void)` (otherwise they accept any number of args).